packages/core/http/core-http-server/src/router/response.ts

```diff
@@ -22,6 +22,41 @@ export interface HttpResponseOptions {
   bypassErrorFormat?: boolean;
 }
 
+/**
+ * HTTP response options for file() responses
+ * @public
+ */
+export interface KibanaFileResponseOptions {
+  /**
+   * HTTP Headers to be sent with the response
+   */
+  headers?: ResponseHeaders;
+  /**
+   * Set to true in order to enable caching for immutable URLs. This means that the
+   * response for the current URL will never change and there isn't any point in the
+   * user requesting the asset again, it should just be cached forever. Only do this
+   * for URLs which include highly-specific information about the response, like hashes
+   * or version numbers, as there is no way to clear that caches on user's machines
+   */
+  immutable?: boolean;
```

Comment:
Taking into account how our averages fellow Kibana contributors are, this option scares me. Especially given the serving endpoint will likely just *not* includes the unique build number or even the version...